website/main/content/blog/2025-06-30-sector-expiration.md

```diff
@@ -0,0 +1,213 @@
+  This `BASE_RANGE` determines the baseline history range for all plots. This `BASE_RANGE`
+  determines the size of history sizes that plots can commit to initially. The plot offset modifies
+  the `BASE_RANGE` to create a unique range for each plot ID, and as the history of the chain grows,
+  this is expanded accordingly so it can fit new history sizes. The following image belongs to a
+  [simple visualisation tool](./2025-06-30-sector-expiration/range-evolution-visual.html) that an
+  LLM has built for me to visualise how the base range and effective range of a plot behaves as the
+  history of the chain grows:
+
+<p align="center">
+<img alt="Range evolution visualisation" src="range_visualisation.png">
+</p>
+
+- Sectors expire as they are used to in current Subspace protocol, so even if the effective range
+  for plots grow, sectors committed to old history sizes will be alreaedy expired (or close to
+  expire) still limiting the number of history sizes that sectors can commit to at a given time.
+  This is key to ensure that farmers do not need to re-plot their sectors as the history grows, and
+  that they can archive the most recent history without having to re-plot, maintaining all the nice
+  properties for sector expiration of the original Subspace protocol.
+
+### Putting all pieces together
+
+With this, we have all that we needed to finally implement farming in shards:
+
+- Plot IDs are unique and stable, allowing us to allocate space into shards.
+- Sectors are committed to a range of history sizes determined by the plot ID, preventing farmers
+  from gaming shard allocation.
+- Piece selection, sector expiration stay unchanged.
+- Farming stays slightly unchanged, apart from the additional information required to verify that
+  the chunk for the piece in the solution range belongs to a sector bound to a plot currently
+  assigned to the shard.
+
+This is how the verification logic for a solution in a shard would look like:
+
+- Standard Subspace Verification (Unchanged): First, perform all existing Subspace checks
+
+```rust
+// Verify piece offset is valid
+if piece_offset >= max_pieces_in_sector {
+    return Err("Invalid piece offset");
+}
+
+// Check sector expiration
+if sector_is_expired(sector_id, current_history_size) {
+    return Err("Sector has expired");
+}
+
+// Verify the piece is part of blockchain history
+if !verify_piece_inclusion(piece, segment_root, proof) {
+    return Err("Invalid piece proof");
+}
+```
+
+- Plot Identity Verification: Verify that the solution comes from a valid plot:
+
+```rust
+// Step 2.1: Compute expected plot ID from farmer's public key and offset
+expected_plot_id = hash(farmer_public_key || plot_offset)
+
+// Step 2.2: Verify the sector was created with this plot ID
+// The sector_id should incorporate the plot_id
+if !sector_belongs_to_plot(sector_id, expected_plot_id) {
+    return Err("Sector doesn't belong to claimed plot");
+}
```

Comment:
In practice this will be used to derive sector ID and if something doesn't match, it'll blow up somewhere later. We'll not check that sector ID belongs to plot ID as such. In fact there is no sector ID in the solution, it is a derived concept not explicitly present in the solution.